### PR TITLE
Add GUIConsole sample

### PR DESCRIPTION
**Pull request by pingzing**
As mentioned in #274, here's a WPF-centric sample. It creates a window that (for now) draws the raw VT100 codes the console emits, and accepts user input. It's based on @waf's MiniTerm C# sample.

I've also set it up to use the ACCENT_ENABLE_ACRYLICBLURBEHIND property, mostly for fun. However, that's accomplished with the undocumented SetWindowCompositionAttribute function. If that should be removed for the sake of keeping scope simple, let me know :+1:

### CODE REVIEW

**Comment by E3V3A** [comment]
This seem to be a useful and welcome PR. Any good reason why it hasn't been merged?

**Comment by miniksa** [comment]
@E3V3A, no, no good reason. Being understaffed. Forgetting about it. I don't know. Something like that.

I just read through it and it looks pretty good to me. Good enough for a sample/seed to start a project using ConPTY.

@DHowett-MSFT, are you upset by the private API for the `ACCENT_ENABLE_ACRYLICBLURBEHIND` being in our repo? I'm not personally upset by it, but I can see why management might crow about us demonstrating a non-published API usage in our repo. I think that'd really be the only sticking point.

**Comment by DHowett-MSFT** [comment]
We should probably err on the side of caution with our partners’ APIs. Thanks!

**Comment by miniksa** [comment]
@pingzing, can you pull the private API usage from your sample then?

I'm honestly currently pursuing a different mechanism by which the acrylic can be accessed in a public and supportable fashion with the team that owns it. There's a thread in my inbox about it. When/if I figure it out, I can try to describe/contribute that back at a later date.

**Comment by pingzing** [comment]
Sure, that's reasonable. I'll yank it and send up the updated version. Should have time for it this weekend or so.

**Comment by miniksa** [comment]
Great, thanks.

**Comment by pingzing** [comment]
@miniksa There we go, all removed. Looks like I also added a title bar at some point, so I decided to throw that in too.

**Comment by waf** [comment]
@DHowett-MSFT's review caught my eye. From a quick look at this PR, it's copy/pasting the source code from https://github.com/Microsoft/console/tree/master/samples/ConPTY/MiniTerm which is resulting in code that is not applicable.

Does it make sense to not copy that code? If we do any bug fixes of the original sample, we'll need to copy the changes to this sample.

**Comment by pingzing** [comment]
Talked to @DHowett-MSFT face-to-face and made some fixes. I think that covers the requested changes, but I should probably go through and remove some the PInvoke wrappers that are now unused.

And @waf, it's _heavily based on_ the MiniTerm sample, but not a perfect 1-1 clone. I made a few architectural changes to make Terminal.cs expose some of what it was doing, so that the WPF app could keep an eye on it. I think we're okay if they diverge over time.

**Comment by pingzing** [comment]
There, removed the now-unused native calls. I'm happy with this if you guys are.

File: samples/ConPTY/GUIConsole/GUIConsole.ConPTY/PseudoConsole.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+        internal static PseudoConsole Create(SafeFileHandle inputReadSide, SafeFileHandle outputWriteSide, int width, int height)
+        {
+            var createResult = CreatePseudoConsole(
+                new COORD { X = (short)width, Y = (short)height },
+                inputReadSide, outputWriteSide,
+                0, out IntPtr hPC);
+            if (createResult != 0)
+            {
+                throw new Win32Exception(createResult, "Could not create psuedo console.");
```
Comment (flug) [inline samples/ConPTY/GUIConsole/GUIConsole.ConPTY/PseudoConsole.cs]: Didn't you mean "pseudo"?

File: samples/ConPTY/GUIConsole/GUIConsole.WPF/Properties/Resources.resx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  <resheader name="version">
+    <value>2.0</value>
+  </resheader>
+  <resheader name="reader">
+    <value>System.Resources.ResXResourceReader, System.Windows.Forms, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <resheader name="writer">
+    <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+</root>
```
Comment (flug) [inline samples/ConPTY/GUIConsole/GUIConsole.WPF/Properties/Resources.resx]: Cariage return ?

File: samples/ConPTY/GUIConsole/GUIConsole.WPF/Properties/Settings.settings [hunk 3]
```diff
@@ -0,0 +1,7 @@
+﻿<?xml version='1.0' encoding='utf-8'?>
+<SettingsFile xmlns="uri:settings" CurrentProfile="(Default)">
+  <Profiles>
+    <Profile Name="(Default)" />
+  </Profiles>
+  <Settings />
+</SettingsFile>
```
Comment (flug) [inline samples/ConPTY/GUIConsole/GUIConsole.WPF/Properties/Settings.settings]: Cariage return ?

**Comment by miniksa** [comment]
@msftbot, make sure @DHowett-MSFT signs off

**Comment by miniksa** [comment]
@pingzing, can you clean up the last few items from @flug's review?

**Comment by miniksa** [comment]
@msft-bot make sure @DHowett-MSFT signs off

**Comment by miniksa** [comment]
@msftbot make sure @DHowett-MSFT signs off

**Comment by pingzing** [comment]
@miniksa Done.

**Comment by miniksa** [comment]
@pingzing, cool, it's late on a Friday and I'm not supposed to really be working anyway. So probably expect Dustin to get back around to this on Monday (if he sees it in the flood of mail we get these days). Sometimes Dustin does an odd or end off hours or on a weekend, but not always. I'll mark it for attention.

**Comment by pingzing** [comment]
@DHowett-MSFT Happy to contribute! Been a pleasure working with all of you.

**Pull request merged**